pkg/daemon/daemon.go

```diff
@@ -2054,6 +2084,13 @@ func (dn *Daemon) prepUpdateFromCluster() (*updateFromCluster, error) {
 		return nil, err
 	}
 
+	if odc == nil {
```

Comment:
Question: why add this here? I think if the on-disk config is empty, this function should be fine, and we shouldn't make assumptions about the current state.

Reply:
This makes sense as well! I originally add the substitution in because I see there are comparisons between the desired vs. current down there. When the current on disk is missing, I think the nearest we can find is the one read from the node annotation, so I substitute that in so that it can make the comparison based on the closest current state we can find for it. But I think what you pointed out ("we should not make assumptions about the current state") really makes sense! And if we leave the current nil, the if statements can still judge on its own. I will take this out!

Reply:
Taken another look at the other two places that calls the odc substitution (now called the getCurrentConfigFromNode) , they are called at steady states (before/after update), where assuming the missing on-disk currentConfig = the current config reading from node annotation is relatively safe.

Reply:
Took another look due to the panic Sergio has reported. I think it is necessary and reasonable to assume odc = current config reading from node annotation if the odc is missing from the disk. It is necessary because below, there are several calls that de-reference the odc to get the currentImage and compares it to the desiredImage to determine whether an update is required. As a result, in the case that odc = nil, de